lightning/src/routing/router.rs

```diff
@@ -957,16 +957,23 @@ impl<'a> CandidateRouteHop<'a> {
 }
 
 #[inline]
-fn max_htlc_from_capacity(capacity: EffectiveCapacity, max_channel_saturation_power_of_half: u8) -> u64 {
+fn max_htlc_from_route_candidate(candidate: &CandidateRouteHop, max_channel_saturation_power_of_half: u8) -> u64 {
 	let saturation_shift: u32 = max_channel_saturation_power_of_half as u32;
-	match capacity {
-		EffectiveCapacity::ExactLiquidity { liquidity_msat } => liquidity_msat,
-		EffectiveCapacity::Infinite => u64::max_value(),
-		EffectiveCapacity::Unknown => EffectiveCapacity::Unknown.as_msat(),
-		EffectiveCapacity::MaximumHTLC { amount_msat } =>
-			amount_msat.checked_shr(saturation_shift).unwrap_or(0),
-		EffectiveCapacity::Total { capacity_msat, htlc_maximum_msat } =>
-			cmp::min(capacity_msat.checked_shr(saturation_shift).unwrap_or(0), htlc_maximum_msat),
+	match candidate {
+		CandidateRouteHop::FirstHop { details } => details.next_outbound_htlc_limit_msat,
+		CandidateRouteHop::PublicHop { info, .. } => {
+			match info.effective_capacity() {
+				EffectiveCapacity::MaximumHTLC { amount_msat } =>
+					amount_msat.checked_shr(saturation_shift).unwrap_or(0),
+				EffectiveCapacity::Total { capacity_msat, htlc_maximum_msat } =>
+					cmp::min(capacity_msat.checked_shr(saturation_shift).unwrap_or(0), htlc_maximum_msat),
+				_ => {
+					debug_assert!(false);
+					EffectiveCapacity::Unknown.as_msat()
+				}
+			}
+		},
+		CandidateRouteHop::PrivateHop { hint } => hint.htlc_maximum_msat.unwrap_or(u64::max_value()),
```

Comment:
nit: `u64::max_value` -> `EffectiveCapacity::Unknown.as_msat()`?

Reply:
Outdated now, but we assume that channels provided in BOLT 11 route hints have sufficient capacity for the entire payment